File: sdks/java/src/test/java/com/echoagent/sdk/MemoryPolicyValuesTest.java
package com.echoagent.sdk;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

class MemoryPolicyValuesTest {
    @Test
    void memoryTypePreservesStabilityAndEligibility() {
        assertEquals(0.85, MemoryType.USER_PREFERENCE.defaultStability());
        assertEquals(0.1, MemoryType.DEPRECATED_NOTE.defaultStability());
        assertTrue(MemoryType.PROJECT_FACT.isRuleEligible());
        assertFalse(MemoryType.WORKFLOW_PATTERN.isRuleEligible());
        assertTrue(MemoryType.DEBUGGING_LESSON.isSkillEligible());
        assertFalse(MemoryType.USER_PREFERENCE.isSkillEligible());
    }

    @Test
    void memorySourcePreservesConfidenceAndRecallWeight() {
        assertEquals(1.0, MemorySource.EXPLICIT_SAVE.defaultConfidence());
        assertEquals(0.6, MemorySource.AUTO_EXTRACTED.defaultConfidence());
        assertEquals(0.9, MemorySource.USER_CORRECTION.defaultRecallWeight());
        assertEquals(0.4, MemorySource.L3_PROMOTION.defaultRecallWeight());
    }
}

File: sdks/typescript/test/memory-policy-values.test.js
import test from "node:test";
import assert from "node:assert/strict";
import {
  MemorySource,
  MemoryType,
  memorySourceDefaultConfidence,
  memorySourceDefaultRecallWeight,
  memoryTypeDefaultStability,
  memoryTypeIsRuleEligible,
  memoryTypeIsSkillEligible,
} from "../dist/index.js";

test("memory type policies preserve Rust stability and promotion eligibility", () => {
  assert.equal(memoryTypeDefaultStability(MemoryType.UserPreference), 0.85);
  assert.equal(memoryTypeDefaultStability(MemoryType.DeprecatedNote), 0.1);
  assert.equal(memoryTypeIsRuleEligible(MemoryType.ProjectFact), true);
  assert.equal(memoryTypeIsRuleEligible(MemoryType.WorkflowPattern), false);
  assert.equal(memoryTypeIsSkillEligible(MemoryType.DebuggingLesson), true);
  assert.equal(memoryTypeIsSkillEligible(MemoryType.UserPreference), false);
});

test("memory source policies preserve confidence and recall weights", () => {
  assert.equal(memorySourceDefaultConfidence(MemorySource.ExplicitSave), 1);
  assert.equal(memorySourceDefaultConfidence(MemorySource.AutoExtracted), 0.6);
  assert.equal(memorySourceDefaultRecallWeight(MemorySource.UserCorrection), 0.9);
  assert.equal(memorySourceDefaultRecallWeight(MemorySource.L3Promotion), 0.4);
});
